### PR TITLE
octave-x: enable qt gui

### PR DESCRIPTION
**Pull request by licy183**
Ref: #21868

### CODE REVIEW

**Comment by twaik** [comment]
Octave in tur is built with custom toolchain so it can be built for 32 bits too.

**Review by twaik — COMMENTED** [review]
In the case if you flip the gui toggle you should fix description because it mentions 'only cli'.

**Comment by licy183** [comment]
I will merge it in 48 hours if there is no more review or when it gets 2+ approves. Thanks!

**Comment by twaik** [comment]
Should not you mention `qt6-qttools-cross-tools` in build dependencies in the case if regular `qt6-qttools` is used ot it is not required?

**Review by twaik — APPROVED** [review]

**Comment by licy183** [comment]
> Should not you mention `qt6-qttools-cross-tools` in build dependencies in the case if regular `qt6-qttools` is used it is not required?

It is better to add it to build dependencies. But actually our building system will always fetch **full** dependencies, so if `qt6-qttools` is added in dependencies, `qt6-qttools-cross-tools` will be fetched too.

File: x11-packages/octave-x/build.sh [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 TERMUX_PKG_LICENSE="GPL-3.0"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="9.2.0"
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_REVISION=2
 TERMUX_PKG_SRCURL=https://ftpmirror.gnu.org/octave/octave-${TERMUX_PKG_VERSION}.tar.xz
 TERMUX_PKG_SHA256=21417afb579105b035cac0bea09201522e384893ae90a781b8727efa32765807
-TERMUX_PKG_DEPENDS="libandroid-complex-math, arpack-ng, bzip2, fftw, fltk, fontconfig, freetype, glpk, glu, graphicsmagick, libcurl, libhdf5, libiconv, libopenblas, libsndfile, opengl, openssl, pcre, portaudio, qhull, qrupdate-ng, rapidjson, readline, suitesparse, sundials, zlib"
-TERMUX_PKG_BUILD_DEPENDS="gnuplot, less"
+TERMUX_PKG_DEPENDS="libandroid-complex-math, arpack-ng, bzip2, fftw, fltk, fontconfig, freetype, glpk, glu, graphicsmagick, libcurl, libhdf5, libiconv, libopenblas, libsndfile, opengl, openssl, pcre, portaudio, qhull, qrupdate-ng, qt6-qtbase, qt6-qttools, qt6-qt5compat, rapidjson, readline, suitesparse, sundials, zlib"
```
Comment (truboxl) [inline x11-packages/octave-x/build.sh:9]:
```suggestion
TERMUX_PKG_DEPENDS="arpack-ng, fftw, fltk, fontconfig, freetype, glpk, glu, graphicsmagick, libandroid-complex-math, libbz2, libc++, libcurl, libhdf5, libiconv, libopenblas, libsndfile, libx11, libxcursor, libxext, libxfixes, libxft, libxinerama, libxrender, opengl, openssl, pcre2, portaudio, qhull, qrupdate-ng, qt6-qt5compat, qt6-qtbase, qt6-qttools, readline, suitesparse, sundials, zlib"
```

File: x11-packages/octave-x/build.sh [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+TERMUX_PKG_BUILD_DEPENDS="gnuplot, less, qt6-qtbase-cross-tools, qt6-qttools-cross-tools"
```
Comment (truboxl) [inline x11-packages/octave-x/build.sh:10]:
```suggestion
TERMUX_PKG_BUILD_DEPENDS="gnuplot, less, rapidjson, qt6-qtbase-cross-tools, qt6-qttools-cross-tools"
```

**Pull request merged**